File: src/java/ArrayStack.java
public class ArrayStack
{
	public String[] stack = new String[10];
	public int p = 0;

	public static void main(String[] args)
	{ 
		ArrayStack p = new ArrayStack();

		p.push("to"); 
		p.push("be"); 
		p.push("or"); 
		p.push("not"); 
		p.push("to"); 

		System.out.println(p.pop()); 

		p.push("be"); 

		System.out.println(p.pop()); 
		System.out.println(p.pop()); 

		p.push("that"); 

		System.out.println(p.pop()); 
		System.out.println(p.pop()); 
		System.out.println(p.pop()); 

		p.push("is"); 
	} 

	public void push(String s)
	{
		stack[p++] = s;
	}

	public String pop()
	{
		return stack[--p];
	}

	public boolean isEmpty()
	{
		return p == 0;
	} 
}

File: src/cs/array_stack.cs
using System; 

class Program
{
	int p = 0; 
	string[] stack = new string[10]; 

	static void Main()
	{
		Program p = new Program(); 

		p.Push("to"); 
		p.Push("be"); 
		p.Push("or"); 
		p.Push("not"); 
		p.Push("to"); 

		Console.WriteLine( p.Pop()); 

		p.Push("be"); 

		Console.WriteLine(p.Pop()); 
		Console.WriteLine(p.Pop()); 

		p.Push("that"); 

		Console.WriteLine(p.Pop()); 
		Console.WriteLine(p.Pop()); 
		Console.WriteLine(p.Pop()); 

		p.Push("is"); 
	}

	public void Push(string s)
	{
		stack[p++] = s; 
	}

	public string Pop()
	{
		return stack[--p]; 
	}

	public bool IsEmpty()
	{
		return p == 0; 
	} 
} 
